pkg/controller/endpointslice/utils.go

```diff
@@ -201,6 +201,17 @@ func objectRefPtrChanged(ref1, ref2 *corev1.ObjectReference) bool {
 	return false
 }
 
+// ownedBy returns true if the provided EndpointSlice is owned by the provided
+// Service.
+func ownedBy(endpointSlice *discovery.EndpointSlice, service *corev1.Service) bool {
+	for _, ownerRef := range endpointSlice.OwnerReferences {
+		if ownerRef.UID == service.UID {
```

Comment:
check the kind and apiVersion as well

Reply:
Updated, thanks! Somehow I'd thought that UID was actually a UUID.